server.go

```diff
@@ -149,6 +150,30 @@ func CustomCodec(codec Codec) ServerOption {
 	}
 }
 
+// CustomCodecs returns a ServerOption that sets the codecs that can be used based on the
+// content-type in addition to the proto codec.
+//
+// TODO: Function name is not amazing
+// TODO: better documentation on overlap with CustonCodec
+func CustomCodecs(codecs ...Codec) ServerOption {
+	m := map[string]Codec{
+		"proto": protoCodec{},
+	}
+	for _, codec := range codecs {
+		// TODO: do we want to do string operations?
+		// I believe TrimSpace does not make a copy, but ToLower does
+		// This might be an extra allocation that is not generally wanted
+		// This matters more in getCodec than here
+		name := strings.TrimSpace(strings.ToLower(codec.String()))
```

Comment:
the current codec validation does not use string operations. does the spec indicate that the content type is case insensitive?